modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/bundle/BatchEngineBundleTracker.java

```diff
@@ -158,55 +139,19 @@ public Bundle addingBundle(Bundle bundle, BundleEvent bundleEvent) {
 				}
 			}
 
-			_batchEngineUnitProcessor.processBatchEngineUnits(
-				singleCompanyBatchEngineUnits);
+			boolean alreadyProcessed = _isAlreadyProcessed(bundle);
+
+			if (!alreadyProcessed) {
+				_batchEngineUnitProcessor.processBatchEngineUnits(
```

Comment:
To simplify, maybe we could create a method registerBatchEngineUnits for the current company instead and mantain a similar flow as in the multiCompany case. Tomorrow we could discuss it, because to me it is a bit confusing that the singleCompanyBatchEngineUnits are not being registered in the map

Reply:
Nevermind. I make some tests and it does not make sense